### PR TITLE
fix: pagination: bug fixes for simplified pagination

### PR DESCRIPTION
**Pull request by rnori-eightfold**
## SUMMARY:

- Remove extra padding on the top in case of simplified pagination
- Fix total prop description
- Show simplified pagination only when there is more than one page
- Use `pageSizes` prop only when `layout` has `PaginationLayoutOptions.Sizes`
- Fix usage of Pagination in Carousel component

## JIRA TASK (Eightfold Employees Only):

N/A

## CHANGE TYPE:

- [x] Bugfix Pull Request
- [ ] Feature Pull Request

## TEST COVERAGE:

- [x] Tests for this change already exist
- [ ] I have added unittests for this change

## TEST PLAN:

Pull the PR branch and run `yarn`, `yarn storybook`. Verify all types of Pagination, Carousel behave as expected.

### CODE REVIEW

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -388,23 +392,25 @@ export const Pagination: FC<PaginationProps> = React.forwardRef(
...
-                      }
-                      quickPreviousIconButtonAriaLabel={
-                        quickPreviousIconButtonAriaLabel
-                      }
-                      simplified={true}
-                      showLast={
-                        !layout.includes(PaginationLayoutOptions.NoLast)
-                      }
-                    />
+                    moreThanOnePage && (
```
Comment (dkilgore-eightfold) [inline src/components/Pagination/Pagination.tsx:395]: We may wish to make this opt-in for now as the `Pagination` component has been displaying when only 1 page for quite a few cycles now. Perhaps with a new boolean prop like `hideWhenSinglePage`? Then default to `false` until we might coordinate the change?

File: src/components/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -642,7 +642,7 @@ export const Carousel: FC<CarouselProps> = React.forwardRef(
                     onCurrentChange={(currentPage: number) =>
                       handleIndicatorClick(currentPage - 1)
                     }
-                    pageSizes={[1]}
+                    pageSize={1}
```
Comment (dkilgore-eightfold) [inline src/components/Carousel/Carousel.tsx:645]: You'll need to ensure this change works both with `pageSizes` and `pageSize` props to ensure no existing implementations are broken.

File: src/components/Pagination/Pagination.stories.tsx [hunk 3]
```diff
@@ -108,7 +108,6 @@ const paginationArgs: Object = {
     PaginationLayoutOptions.Next,
   ],
   pageSize: 10,
-  pageSizes: [10, 20, 30, 40, 50, 100],
```
Comment (dkilgore-eightfold) [inline src/components/Pagination/Pagination.stories.tsx:111]: Same as above. The component functionality needs to remain the same, but with your fix as an improvement.

File: src/components/Pagination/pagination.module.scss [hunk 4]
```diff
@@ -95,7 +95,6 @@
 
   .page-tracker {
     opacity: 50%;
-    padding-top: $space-xxs;
```
Comment (dkilgore-eightfold) [inline src/components/Pagination/pagination.module.scss:98]: This should be done in the HOC, remove.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Carousel/Carousel.tsx (hunk 2 above)
Comment (rnori-eightfold) [reply src/components/Carousel/Carousel.tsx:645]: This specific change is harmless as its a change within the library. But I get the gist of the comment.

File: src/components/Pagination/Pagination.tsx (hunk 1 above)
Comment (rnori-eightfold) [reply src/components/Pagination/Pagination.tsx]: Agree, added `hideWhenSinglePage` prop as suggested

File: src/components/Pagination/Pagination.stories.tsx (hunk 3 above)
Comment (rnori-eightfold) [reply src/components/Pagination/Pagination.stories.tsx:111]: Reverted this change

File: src/components/Carousel/Carousel.tsx (hunk 2 above)
Comment (dkilgore-eightfold) [reply src/components/Carousel/Carousel.tsx:646]: Tested both using `Carousel.tsx` and LGTM

File: src/components/Pagination/Pagination.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Pagination/Pagination.tsx]: @rnori-eightfold You should update `Pager` to display the `pageCount` when `!hideWhenSinglePage`, currently its compare only is relevant to `!simplified` scenarios. Perhaps keep the same condition and breakout `simplified` to its own conditional `<li>` using `simplified && pageCount > 0 && showLast` as the condition? 

https://github.com/rnori-eightfold/octuple/blob/rnori-eightfold/simplified-pager-bug-fixes/src/components/Pagination/Pager.tsx#L230

Comment (rnori-eightfold) [reply src/components/Pagination/Pagination.tsx]: @dkilgore-eightfold Thanks for pointing out. Pushed a [fix](https://github.com/EightfoldAI/octuple/pull/581/commits/1b5d2f25a46f7f5cb0c83ceb9d0a6a1a4eb34306) for this

File: src/components/Pagination/Pager.tsx [hunk 5]
```diff
@@ -227,7 +227,7 @@ export const Pager: FC<PagerProps> = React.forwardRef(
               />
             </li>
           )}
-        {pageCount > 1 && showLast && (
+        {!simplified && pageCount > 1 && showLast && (
           <li>
             {!simplified ? (
```
Comment (dkilgore-eightfold) [inline src/components/Pagination/Pager.tsx:232]: nested ternary may be removed.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Pagination/Pager.tsx [hunk 6]
```diff
@@ -227,25 +227,26 @@ export const Pager: FC<PagerProps> = React.forwardRef(
...
+              size={ButtonSize.Medium}
+              text={pageCount.toLocaleString()}
+            />
+          </li>
+        )}
+        {simplified && pageCount > 0 && showLast && (
+          <li>
+            <span>{pageCount.toLocaleString()}</span>
           </li>
         )}
```
Comment (dkilgore-eightfold) [inline src/components/Pagination/Pager.tsx:251]: Remove.

```suggestion
```

Then update lines 133-158 to:

```
      {pageCount > 0 && (
          <li>
            {!simplified ? (
              <NeutralButton
                aria-current={currentPage === 1}
                classNames={mergeClasses([
                  styles.paginationButton,
                  { [styles.active]: currentPage === 1 },
                ])}
                shape={ButtonShape.Rectangle}
                onClick={() => onCurrentChange(1)}
                size={ButtonSize.Medium}
                text={'1'.toLocaleString()}
              />
            ) : (
              <>
                <span>{`${currentPage.toLocaleString()} ${
                  locale.lang!.pagerText
                }`}</span>{' '}
                {pageCount > 0 && showLast && (
                  <span>{`${pageCount.toLocaleString()}`}</span>
                )}
              </>
            )}
          </li>
        )}
```

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

**Review by dkilgore-eightfold — APPROVED** [review]

**Comment by ranjithnori** [comment]
:shipit:

**Pull request merged**